### PR TITLE
Windows Build: Add notes for multi-line commands when using PowerShell

### PR DESCRIPTION
**Pull request by MikeChongCan**
On Windows, we have two shells:
- Command Prompt
- PowerShell

The use of PowerShell is increasing. Since all the commands of this file is written in Command Prompt, we need to add a note for whose who do not know this difference.

### CODE REVIEW

File: docs/WindowsBuild.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Visual Studio 2017 or newer is needed to build Swift on Windows. The free Commun
 
 The commands below (with the exception of installing Visual Studio) must be entered in the "**x64 Native** Tools Command Prompt for VS2017" (or VS2019, VS2019 Preview depending on the Visual Studio that you are using) in the Start Menu. This sets environment variables to select the correct target platform.
 
+All the commands below are written in Command Prompt. While using PowerShell, You need to replace `^` with <code>`</code> (backtick) at the end of each line.
```
Comment (imWildCat) [inline docs/WindowsBuild.md:7]: The following markdown code does not work on Github:
```md
`\``
```
It will show:
`\``

**Comment by compnerd** [comment]
I'm not opposed to the change, but would like to understand how that confusion comes about as Microsoft does not provide a x64 PowerShell Developer Command Prompt AFAIK (or at least did not when I last installed VS).  Has that changed since?

**Comment by imWildCat** [comment]
@compnerd Thanks for your question!

If my understanding is correct, users are likely to user PowerShell by default so there would be confusion. According to [this documentation](https://docs.microsoft.com/en-us/powershell/scripting/windows-powershell/install/installing-windows-powershell?view=powershell-7.1),
> Windows PowerShell comes installed by default in every Windows, starting with Windows 7 SP1 and Windows Server 2008 R2 SP1.

In addition, in popular terminal apps such as [Windows Terminal](https://docs.microsoft.com/en-us/windows/terminal/), PowerShell should be the default one.


(I am not working on this as an employer and my primary working machine is MacBook. Just thinking of this while I was trying to compile Swift on my personal computer)

**Comment by compnerd** [comment]
@imWildCat even better :) I think that having clear instructions to enable others to easily work on the project is important.

Your clarification relates more towards a justification for using PowerShell.  I'm not really tied to cmd or PowerShell.  However, the thing is, the changes to build with PowerShell here are going to be insufficient.  The commands that are listed further along are dependent on the handling of variables in CMD as well as the default environment variables setup by VsDevEnv, which gives you CMD.

Is there a way to get a x64 Developer PowerShell?

**Comment by imWildCat** [comment]
@compnerd You're right. I did a bit research and cannot find a good way to use `VsDevEnv` in PowerShell.

Moreover, `mklink` commands also need to be replaced with PowerShell alternatives.

Given this fact, I am closing this PR now.

**Pull request closed without merging**